### PR TITLE
bowtie2: import from homebrew/science

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

https://github.com/Homebrew/homebrew-science/pull/6345

* removed option for TBB since it only fixed a Linux problem https://github.com/Homebrew/homebrew-science/pull/2586
* the `SDKROOT` issue on el cap is due to https://github.com/Homebrew/homebrew-science/pull/5602
* updated the test for the fancy new `assert_predicate`

### CODE REVIEW

**Comment by ilovezfs** [comment]
>the SDKROOT issue on el cap is due to Homebrew/homebrew-science#5602

Unfortunately this will mean it cannot build with Xcode alone, which needs to be fixed upstream before this can be merged.

**Comment by jonchang** [comment]
@ilovezfs hmm, since that line fixes an issue that only affects tophat (which we aren't migrating), I could probably just delete it.

**Comment by ilovezfs** [comment]
:+1: to that

**Comment by jonchang** [comment]
* removed tophat-specific fix

File: Formula/bowtie2.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Bowtie2 < Formula
+  desc "Fast and sensitive gapped read aligner"
+  homepage "https://bowtie-bio.sourceforge.io/"
+  url "https://github.com/BenLangmead/bowtie2/archive/v2.3.3.tar.gz"
+  sha256 "7d4d455a4af70d02a1ae4ce1a0b21b7d3018737dd197579e1a5612a5c01887c8"
+  head "https://github.com/BenLangmead/bowtie2.git"
```
Comment (ilovezfs) [inline Formula/bowtie2.rb]: let's remove head

**Comment by jonchang** [comment]
* head is dead

**Comment by jonchang** [comment]
I will see about getting analytics access so you don't have to keep checking for me 😶

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks @jonchang!

**Pull request closed without merging**